File: course CS50/6 python/cash.py
#! python3
""" CS50 PSet 6: Cash
    Calculates the minimum number of coins required to give a user change. """

answer = 0.0
while answer <= 0:
    try:
        answer = float(input("Change owed: "))
    except ValueError:
        continue

change = round(answer * 100)

coins = 0
while change > 0:
    if change // 25 >= 1:
        change -= 25
        coins += 1

    elif change // 10 >= 1:
        change -= 10
        coins += 1

    elif change // 5 >= 1:
        change -= 5
        coins += 1

    else:
        change -= 1
        coins += 1

print(coins)

File: course CS50/1 C/cash.c
/*  fall 2020 CS50 PSet1: Cash
 *
 *  Ask the user for the change owed,
 *  count minimum number of coins needed to pay.
 */

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int cash(int);

int main(void)
{
    float answer;
    do
    {
        answer = get_float("Change owed: ");
    }
    while (answer <= 0);

    int change = round(answer * 100);
    int coins = cash(change);

    printf("%i\n", coins);

    return 0;
}

int cash(int change)
{
    int coins = 0;
    while (change > 0)
    {
        if (change / 25 >= 1)
        {
            change -= 25;
            coins++;
        }
        else if (change / 10 >= 1)
        {
            change -= 10;
            coins++;
        }
        else if (change / 5 >= 1)
        {
            change -= 5;
            coins++;
        }
        else
        {
            change -= 1;
            coins++;
        }
    }
    return coins;
}
